portality/tasks/reporting.py

```diff
@@ -88,6 +88,21 @@ def _tabulate_time_entity_group(group, entityKey):
                     existing = True
             if not existing:
                 table.append([u] + padding + [c])
+
+        # Add a 0 for each user who has been added to the table but doesn't
+        # have any actions in the current time period we're looping over. E.g.
+        # if we're counting edits by month, this would be "users who were active
+        # in a previous month but haven't made any edits this month".
+        users_in_table = set(map(lambda each_row: each_row[0], table))
+        previously_active_users = users_in_table - set(users_active_this_period)
+        for u in previously_active_users:
```

Comment:
Couldn't this be simplified to:

```python
for row in table:
    if row[0] in previously_active_users:
        row.append(0)
```
i.e. we shouldn't need to loop through the rows for each user, just through the rows once.